### PR TITLE
Add `ThemeProvider`

### LINKED ISSUE #47: Add `ThemeProvider`
There is a POC of `ThemeProvider` usage on branch `kamil.theme-provider`. An example of how to use the component should be added as well.

### PR DESCRIPTION
**Pull request by jakubsob**
Closes #47 

Changes proposed in this pull request:
- Added `ThemeProvider` 

# PR checklist

- [x] E2E tests added
- [x] Documentation updated

### CODE REVIEW

File: inst/examples/e2e-test/R/themeProvider.R [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    neutralDark = "#f4f4f4",
+    black = "#f8f8f8",
+    white = "#323130"
+  )
+)
+
+themeProviderUI <- function(id) {
+  ns <- NS(id)
+  div(
+    `data-cy` = ns("themeProviderContent"),
```
Comment (kamilzyla) [inline inst/examples/e2e-test/R/themeProvider.R:31]: I don't mind this solution but is there a reason to use `data-cy` instead of `id` in tests?

**Review by kamilzyla — APPROVED** [review]
Awesome, thanks! Please also add the example to `inst/examples/dashboard/examples.R` and verify that it shows when you run `shiny.fluent::runExample("dashboard")`. Alternatively create a separate issue for this task.

File: inst/examples/e2e-test/R/themeProvider.R (hunk 1 above)
Comment (jakubsob) [reply inst/examples/e2e-test/R/themeProvider.R:31]: Using `data-cy` is a suggested solution by Cypress itself so I've decided to take advantage of it

**Comment by jakubsob** [comment]
@kamilzyla This is a ThemeProvider page in `dashboard` example, PTAL
image

**Comment by kamilzyla** [comment]
LGTM, feel free to merge once the CI passes! :tada:

**Pull request merged**